dev/tasks/r/github.packages.yml

```diff
@@ -375,29 +375,30 @@ jobs:
           read_parquet(system.file("v0.7.1.parquet", package = "arrow"))
           print(arrow_info())
 
-  #TODO test macos source build?
-  test-linux-source:
+  test-source:
     needs: source
-    name: Test linux source build
-    runs-on: ubuntu-latest
+    name: Test {{ '${{ matrix.platform.name }}' }} source build

```

Comment:
What is the difference with this "Test macOS source build" and "r-packages"?
It seems that both of them use pre-built Apache Arrow C++ and build Apache Arrow R from source.

Reply:
The macos source job builds libarrow and the package from source (see `LIBARROW_BINARY=false` ), same as the previously existing linux job.

Reply:
https://github.com/ursacomputing/crossbow/actions/runs/6490158381/job/17625543388#step:13:42

Reply:
Ah, all Apache Arrow C++ build logs are suppressed. Sorry. You're right.

Can we improve the "Install arrow from nightly repo" step name? It seems that we don't use "nightly repo". (We use "local repo" that has a source package created by the "Source Package" job.)